File: lms.backend/src/main/java/com/lms/backend/service/StudentService.java
// StudentService.java - COMPLETE INTERFACE
package com.lms.backend.service;

import com.lms.backend.model.Course;
import com.lms.backend.model.User;

import java.util.List;
import java.util.Map;

public interface StudentService {

    // Enrollment methods
    void enrollCourse(User student, Long courseId);
    List<Course> getEnrolledCourses(User student);

    // Progress tracking methods
    Map<String, Object> getCourseWithProgress(User student, Long courseId);
    void markLessonCompleted(User student, Long courseId, Long lessonId);
    Map<String, Object> getLearningStats(User student);
    Map<String, Object> getCourseProgress(User student, Long courseId);

    // Course discovery methods
    List<Course> getAvailableCourses(User student);

    List<Course> getCourseCatalog();
}

File: lms.frontend/src/services/studentService.js
import apiClient from './apiClient'; // Use your axios instance

const studentService = {
  
  async enrollCourse(courseId) {
    const response = await apiClient.post(`/student/enroll/${courseId}`);
    return response.data;
  },

  async getEnrolledCourses() {
    const response = await apiClient.get('/student/my-courses');
    return response.data;
  },

  async getAvailableCourses() {
    const response = await apiClient.get('/student/courses/available');
    return response.data;
  },

  async getCourseWithProgress(courseId) {
    const response = await apiClient.get(`/student/course/${courseId}`);
    return response.data;
  },

  async markLessonCompleted(courseId, lessonId) {
    const response = await apiClient.post(`/student/course/${courseId}/lesson/${lessonId}/complete`);
    return response.data;
  },

  async getLearningStats() {
    const response = await apiClient.get('/student/stats');
    return response.data;
  }
};

export default studentService;